modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/model/impl/CPDefinitionImpl.java

```diff
@@ -197,6 +202,43 @@ public List<CPAttachmentFileEntry> getCPAttachmentFileEntries(
 			QueryUtil.ALL_POS);
 	}
 
+	@Override
+	public CPConfigurationEntry getCPConfigurationEntry(
+			long cpConfigurationListId)
+		throws PortalException {
+
+		List<CPConfigurationEntry> cpConfigurationEntries =
+			CPConfigurationEntryLocalServiceUtil.getCPConfigurationEntries(
+				ClassNameLocalServiceUtil.getClassNameId(CPDefinition.class),
+				getCPDefinitionId());
+
+		for (CPConfigurationEntry cpConfigurationEntry :
+				cpConfigurationEntries) {
+
+			if (cpConfigurationEntry.getCPConfigurationListId() ==
+					cpConfigurationListId) {
+
+				return cpConfigurationEntry;
+			}
+
+			CPConfigurationEntrySetting cpConfigurationEntrySetting =
+				CPConfigurationEntrySettingLocalServiceUtil.
+					fetchCPConfigurationEntrySetting(
+						cpConfigurationEntry.getCPConfigurationEntryId(),
+						CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+			if ((cpConfigurationEntrySetting != null) &&
+				StringUtil.contains(
+					cpConfigurationEntrySetting.getSetting(),
```

Comment:
how is structured the getSetting string? I mean... with just the contains can we have cases like
```
100,500,600
```
and if I search for just 50 it returns true

Reply:
it contains ids, that are number generated with a sequence, I think it's very unlikely the case you write and even if it happen I think it would end up in some performance loss, it should not break the execution